planner/core/exhaust_physical_plans.go

```diff
@@ -1774,17 +1774,28 @@ func (p *LogicalProjection) TryToGetChildProp(prop *property.PhysicalProperty) (
 }
 
 func (p *LogicalProjection) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]PhysicalPlan, bool) {
```

Comment:
IMO, we don't need to change this function to check if projection can be checked. All the check can be done in `attach2Task`, such as what kind of task it is and whether the expression can be pushed

Reply:
move the limination to `attach2Task`